Additional SAML Documentation (#4587)
mendix/docs: content/en/docs/appstore/modules/saml.md

1. Configure the **Startup** microflow to run as the startup microflow. This microflow will initialize the custom request handler `/SSO/` (please note the importance of using the final `/` for all instances of `/SSO/`), validate all IdP configurations, and prepare the configuration entities required during the configuration.
2. Add the **OpenConfiguration** microflow to the navigation, and then allow the administrator to access this page.
3.  Review and configure all the constants:

> 	* **DefaultLoginPage** – You can specify a different login page here (for example, configuring the `index.html` page to redirect to `/SSO/`). This constant is only used when the login process fails. When the end-user cannot be authenticated in Mendix, they are presented with a page. If this constant is specified, a button will appear, and by clicking this button, you will be redirected to the default login page (for example, `index.html` or `login.html`).

Author: I find this reference to redirecting very confusing.
I think I would try to make it simpler:
```suggestion
	* **DefaultLoginPage** – You can specify a different login page here for when the login process fails. When the end-user cannot be authenticated in Mendix, a button will appear, and by clicking this button, you will be redirected to the specified login page. If this is left blank, an unauthenticated user will be redirected to `/login.html`. 
```

Reviewer: Simpler, but not entirely correct:
> When the end-user cannot be authenticated in Mendix

should be

> When the end-user cannot be authenticated in the external Identity Provider

The idea is that when an end-user fails to sign in with the IdP, they are redirected to the normal login page, so they can try to sign in there, directly in the Mendix app.

Also, I think
>you will be redirected to the specified login page.

should be

>they will be redirected to the specified login page.

?

Author: Thanks Feddo - taking your suggestions into account, does this look OK?
```suggestion
	* **DefaultLoginPage** – You can specify a different login page here for when the login process fails. When the end-user cannot be authenticated in the external Identity Provider, a button will appear, and by clicking this button, they will be redirected to the specified login page. If this is left blank, an unauthenticated user will be redirected to `/login.html`.
```

Reviewer: Yes, looks ok!

If encryption is enabled, all the certificates required for encryption are stored in the key store. When you choose **Use encryption** a key store is automatically created using the URL of the application, or the custom entity Id, and shown as the **Key store alias**. See [Managing the Key Store](#keystore), below, for more key store options.

Accessing the metadata for the SP can be done either by clicking **Download SP Metadata** to download the XML file or by opening `http://<Application Root URL>/SSO/metadata` for your app's URL.

#### 3.1.1 Managing the Key Store{#keystore}

You can usually leave the key store settings as the default. However, there may be a requirement to use a specific key store.

{{% alert color="warning" %}}
Resetting the key store or uploading another key store will require all the IdPs to import the new metadata file. 
{{% /alert %}}

Click **Upload** to upload a key store file. Use the **Entity Id** as the the alias of the key store.

> You can create your own key store file, which you can use when configuring other SAML SPs by clicking **Download**.

Reviewer: Not sure if this is confusing. Clicking the button doesn't create the file: it is already there. It will just download the existing file.

Author: ```suggestion
You can download the key store file and use it when configuring other SAML SPs by clicking **Download**.
```

Reviewer: 👍

{{% alert color="info" %}}
Remember to set the new key store password in the `KeystorePassword` constant of your app.
{{% /alert %}}

To return the key store settings to their defaults, click **Reset**.

### 3.2 Configuring the IdP-Specific Settings

Each IdP (entity descriptor) should have its own configuration set. Every IdP can be configured and enabled separately. All changes made in the configuration are immediately applied when you save the configuration. 

#### 3.2.1 Creating a New IdP Configuration

{{% alert color="warning" %}}
If you have multiple IdPs, please make sure each IdP has a unique **Entity descriptor**. If you add multiple IdPs with the same entity descriptor, you might experience unexepected behavior where a different SSO configuration is selected than the alias provided.
{{% /alert %}}

When creating a new IdP configuration, you are guided through a workflow to help you configure everything required for the IdP configuration. Each option in the workflow is explained below, and can be changed by editing an existing IdP Configuration.

> Upon completing these steps, you only need to send the metadata file to the IdP and have them configure the authentication on their end. 

Reviewer: the _SP_ metadata file

Author: ```suggestion
Upon completing these steps, you only need to send the SP metadata file to the IdP and have them configure the authentication on their end. 
```

Reviewer: 👍

> If you want to automatically synchronize the IdP metadata, go to **\_USE ME** > **Scheduled Events** in the **App Explorer**, double-click **SE_SynchronizeIdPMetadata** and select **Yes** for **Enabled** to enable the automatic synchronization of the IdP metadata. 

Reviewer: This looks weird: changing the _Enabled_ setting of an scheduled event only applies to running locally, or on a sandbox environment. It doesn't enable the scheduled event when running the app in in the cloud. Maybe we should make this more generic?

>If you want to automatically synchronize the IdP metadata, make sure the **SAML20.SE_SynchronizeIdPMetadata** [scheduled event](/refguide/scheduled-events/) is enabled.

Author: I've taken your suggestion and just pointed the user to where to find it.
Should we have a suggestion here on how often to run this, or just leave it to the customer?

```suggestion
If you want to automatically synchronize the IdP metadata, make sure the **SE_SynchronizeIdPMetadata** [scheduled event](/refguide/scheduled-events/) is enabled. This is in the **\_USE ME** > **Scheduled Events** folder of the SAML module.
```

Reviewer: 👍
